### PR TITLE
CDK-973: Add View#asSchema and View#asType projection methods

### PR DESCRIPTION
**Pull request by rdblue**
This extends Joey's PR, #346, and fixes the review issues I noted. One of the main issues was that without changing the backing dataset's type class, projection would fail. This adds `asType` projection to take care of that issue.

### CODE REVIEW

**Comment by rdblue** [comment]
@joey, could you review these updates to your original PR? I went ahead and made the changes because I want to make sure this is in the next release.

File: kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemViewKeyInputFormat.java [hunk 1]
```diff
@@ -71,37 +69,38 @@
...
-            SpecificData.get().getSchema(dataset.getType()).toString());
-      }
-    } else if (Formats.PARQUET.equals(format)) {
-
-      // Use the reader's schema type if provided.
-      if (isSpecific) {
-
-        AvroReadSupport.setAvroReadSchema(conf,
-            SpecificData.get().getSchema(dataset.getType()));
+    if (!DataModelUtil.isGeneric(dataset.getType())) {
```
Comment (rdblue) [inline kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemViewKeyInputFormat.java]: I don't think this should only happen if the data model is not generic. If it is generic, we can still set the data model and schema correctly.

**Comment by joey** [comment]
+1, let me know if you want me to merge this or if you want to handle it.

**Comment by rdblue** [comment]
I'll merge. Thanks for reviewing it!

**Pull request merged**